plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClusteringPredicatesExtractor.java

```diff
@@ -65,102 +67,135 @@ private static ClusteringPushDownResult getClusteringKeysSet(List<CassandraColum
             if (domain.isNullAllowed()) {
                 break;
             }
+
+            int currentlyProcessedClusteringColumn = allProcesedClusteringColumns;
             String predicateString = domain.getValues().getValuesProcessor().transform(
```

Comment:
Can we just use something like
```
if (domain.isSingleValue()) {
   // construct equality predicate from domain.getSingleValue()
}
else if (domain.isNullableDiscreteSet()) {
   // construct IN predicate from domain.getNullableDiscreteSet()
}
else if (type.isOrderable()) {
   // construct range predicate from domain.getValues().getRanges().getSpan()
}
```
In first 2 cases it is full pushdown, in 3rd case it is not

Reply:
I think this would be a safer and easier to understand approach but I'll leave it to @hashhar to decide if we should do it here or in a follow up.

Reply:
That looks like a refactor and I agree it looks much simpler than the current code but let's keep it for a follow-up since the current code is a smaller diff that fixes the original issue.